### PR TITLE
[BUGFIX beta] exempt routes that share a controller from duplicate assertion

### PR DESCRIPTION
**Pull request by ghost**
start of fix for #14560

### CODE REVIEW

**Comment by efx** [comment]
@trentmwillis mind reviewing to ensure fix is in right direction? 
Ideally, a fix for this could be merged since this issue is blocking our upgrade. I figured a PR with the beginnings here is more sensible then writing some workaround specific to our application.

File: packages/ember-routing/lib/system/router.js
Comment (trentmwillis) [inline packages/ember-routing/lib/system/router.js]: I'm unsure the `hasOwnProperty` check here is necessary. Ideally if a collision happens, and the property isn't there then we should do the assertion.

File: packages/ember-routing/tests/system/router_test.js
Comment (trentmwillis) [inline packages/ember-routing/tests/system/router_test.js]: I would rather these tests be done as acceptance-type tests rather than unit tests. They should probably go in this test file which does the other collision checks: https://github.com/emberjs/ember.js/blob/master/packages/ember/tests/routing/query_params_test/overlapping_query_params_test.js

**Review by trentmwillis — CHANGES_REQUESTED** [review]
Thanks for working on this!

**Comment by efx** [comment]
thanks for the feedback; will be updating here within the morning.

File: packages/ember/tests/routing/query_params_test/overlapping_query_params_test.js
Comment (trentmwillis) [inline packages/ember/tests/routing/query_params_test/overlapping_query_params_test.js]: Nitpick: Need a newline before and after the test block

**Review by trentmwillis — CHANGES_REQUESTED** [review]
One small thing, otherwise seems good to me. Will need a review from an actual maintainer though.

**Review by trentmwillis — APPROVED** [review]
Looks good to me!

**Comment by rwjblue** [comment]
OK, I squashed, rebased, and updated commit message.  Will merge once Travis is green...

**Comment by efx** [comment]
thanks @trentmwillis and @rwjblue !

**Pull request merged**
